src/ansari/ansari_db.py

```diff
@@ -109,13 +112,95 @@ def _get_token_from_request(self, request: Request) -> str:
                 detail="Authorization header is malformed",
             )
 
+    def _execute_query(
+        self,
+        query: Union[str, list[str]],
+        params: Union[tuple, list[tuple]],
+        which_fetch: Union[Literal["one", "all"], list[Literal["one", "all"]]] = "",
+        commit_after: Literal["each", "all"] = "each",
+    ) -> Union[tuple, Optional[list[tuple]], list[Optional[list[tuple]]]]:
+        """
+        Executes one or more SQL queries with the provided parameters and fetch types.
+
+        Args:
+            query (Union[str, List[str]]): A single SQL query string or a list of SQL query strings.
+            params (Union[tuple, List[tuple]]): A single tuple of parameters or a list of tuples of parameters.
+            which_fetch (Union[Literal["one", "all"], List[Literal["one", "all"]]]):
+                A single fetch type or a list of fetch types. Each fetch type can be:
+                - "one": Fetch one row.
+                - "all": Fetch all rows.
+                - Any other value: Do not fetch any rows.
+            commit_after (Literal["each", "all"]): Whether to commit the transaction after each query is executed,
+                or only after all of them are executed.
+
+        Returns:
+            Union[Optional[List], List[Optional[List]]]:
```

Comment:
What if we simplified this a bit and made this always return a list (even if there is only one element)? 

I am not sure so this is only a suggestion, but having multiple return types feels slightly clumsy to me. 

<img width="700" alt="image" src="https://github.com/user-attachments/assets/5a1e548e-7d41-4cd6-8ecb-3f0e8cc6402f" />

Reply:
Good point, will adjust it 👍 

Also, I _love_ the attached screenshot; any attempts to explain reasoning/advantages/disadvantages are always appreciated :]